cmake/sysbuild/suit.cmake

```diff
@@ -176,13 +176,19 @@ function(suit_create_package)
   foreach(image ${IMAGES})
     sysbuild_get(BINARY_DIR IMAGE ${image} VAR APPLICATION_BINARY_DIR CACHE)
     sysbuild_get(BINARY_FILE IMAGE ${image} VAR CONFIG_KERNEL_BIN_NAME KCONFIG)
+    sysbuild_get(target IMAGE ${image} VAR CONFIG_SUIT_ENVELOPE_TARGET KCONFIG)
 
     set(BINARY_FILE "${BINARY_FILE}.bin")
 
     list(APPEND CORE_ARGS
       --core ${image},${SUIT_ROOT_DIRECTORY}${image}.bin,${BINARY_DIR}/zephyr/edt.pickle,${BINARY_DIR}/zephyr/.config
     )
+
+    list(APPEND CORE_ARGS
+    --core ${target},${SUIT_ROOT_DIRECTORY}${target}.bin,${BINARY_DIR}/zephyr/edt.pickle,${BINARY_DIR}/zephyr/.config
+    )
```

Comment:
I have one concern here. in `flash_companion` I didn't set any value for `CONFIG_SUIT_ENVELOPE_TARGET`, which means that it has `application` by default. In that case both `smp_transfer` and `flash_companion` would have the same target name, which is a recipe for problems.

I am wondering how `CONFIG_SUIT_ENVELOPE_TARGET` should be approached. On one hand, I think that using `sysbuild` image name by default is convenient and would be easier for the user.

On the second hand, we want the default manifest templates to be generic regardless of the application name. 

Perhaps in the `flash_companion` the `CONFIG_SUIT_ENVELOPE_TARGET` should be set to `"" (empty)` and in this loop there should be a check if there are multiple images with the same `CONFIG_SUIT_ENVELOPE_TARGET` and a warning should be displayed in such case.